### ISSUE TITLE
Add an option to disable caching in SDWebImageManager [discussion]

### ISSUE THREAD

**Issue by KiranPanesar** [rs/SDWebImage#1146]
**tl;dr**:  We want to disable ```SDWebImageManager``` from **writing** to the cache, but we still want to have it **read** from the cache.

---

We are using ```SDWebImage``` to load a large number of large images (about 1MB each). In our initial setup we were using the ```UIImageView+WebCache``` category methods to set the images in ```UICollectionViewCell``` image views. We soon ran into the ```NSCache``` memory leak bug described in #538. The general solution in that thread is to disable memory caching, which isn't really an option.

Our 'solution' is to scale these images down proportionally to be no larger than the screen width and then compress with ```UIImageJPEGRepresentation```. This makes leak unnoticeable and until Apple fixes ```NSCache```, it's as much as we can do whilst still caching to memory.

Right now we have written a custom alternative to the ```UIImageView+WebCache``` category methods that uses ```SDWebImageDownloader``` to download the image and then we perform our manipulation to scale/compress and then cache the result. You can see our WIP method [here](https://gist.github.com/KiranPanesar/3d1ac45e7b8aadf548b3).

We are experiencing an issue with the threading and racing, where incorrect images are being returned when we're running this loading on the background thread. It appears that ```SDWebImageManager``` has ways to handle these issues (using ```@synchronize```). So we can make our wrapper around ```SDWebImageDownloader``` (including our asynchronous loading from disk) thread-safe, but I would much prefer to use the well-constructed ```SDWebImageManager``` to handle reading from the cache and downloading in a thread-safe manner.

So I am proposing adding an option such as ```SDWebImageCacheDisabled``` which would disable the ```SDWebImageManager``` caching, but it would still load the cache if present. Similar to a cache policy. That would then allow us to handle the resizing and compressing and cache it independently.

It would be a very simple feature to add, and we would build it. I would just like to get the core contributors' thoughts on this approach and the likelihood of it getting merged in. I want to avoid building this out on our own fork and then never having it merged in so we are working on an outdated fork in the future. If it's not something you would consider merging it, then please close the issue and we will implement a solution internally (probably through subclassing).

**Comment by KiranPanesar** [comment]
Closing this after discovering the [delegate methods on SDWebImageManager](https://github.com/rs/SDWebImage/blob/master/SDWebImage/SDWebImageManager.h#L120) which allow you to manipulate the ```UIImage``` as soon as it's finished downloading before caching.
